subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoReportsContainer.java

```diff
@@ -27,7 +27,7 @@
  * The reporting configuration for the {@link JacocoReport} task.
  */
 @Incubating
-public interface JacocoReportsContainer extends ReportContainer<Report> {
+public interface JacocoReportsContainer extends ReportContainer<ConfigurableReport> {
```

Comment:
It's OK to have a breaking change it. We'll likely introduce this concept with 4.0.

Changed from `all` to `withType`.